### PR TITLE
Add github workflows

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: .github/workflows/manual-build.yml
Comment (MrCreosote) [inline .github/workflows/manual-build.yml:1]: None of the image build GHA scripts are going to work when it comes to pushing an image

File: .github/dependabot.yml
Comment (MrCreosote) [inline .github/dependabot.yml:1]: Where did this and codeql come from?

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
...
-Flask==1.0.2
-gunicorn==19.9.0
-gevent==1.3.5
\ No newline at end of file
+Flask==2.0.0
+gevent==24.2.1
+gunicorn==22.0.0
+pymongo==3.10
+requests==2.20.0
+Werkzeug==2.0.3
```
Comment (MrCreosote) [inline requirements.txt:7]: Why do we need a Werkzeug install when we didn't before?

File: src/jgikbase/test/idmapping/service/mapper_service_test.py [hunk 2]
```diff
@@ -239,8 +239,8 @@ def test_root_and_logging():
     assert_ms_epoch_close_to_now(time_)
     assert resp.status_code == 200
 
-    assert len(logstream.write.call_args_list) == 2
-    assert logstream.write.call_args_list[1][0][0] == '\n'
+    assert len(logstream.write.call_args_list) == 1
+    assert logstream.write.call_args_list[0][0][0][-1] == '\n'
```
Comment (MrCreosote) [inline src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: Why do the logs change in these tests?

**Comment by MrCreosote** [comment]
BTW what caused the hang problem?

File: .github/dependabot.yml
Comment (Xiangs18) [reply .github/dependabot.yml:1]: from collections. Same as Handle service 2

File: requirements.txt (hunk 1 above)
Comment (Xiangs18) [reply requirements.txt:7]: when install Flask, Werkzeug is automatically installed. 
I initially upgraded Flask to the latest version 3.0.3, which also installed Werkzeug to 3.0. 
However, `werkzeug.server.shutdown` is deprecated and removed in Werkzeug 2.1
So in 3.0, it never properly shutdown the server for each function, causing the hanging problem.
Therefore I manually downgraded Werkzeug to 2.0.3 fixed the problem.

File: .github/workflows/manual-build.yml
Comment (Xiangs18) [reply .github/workflows/manual-build.yml:1]: We are switching to https://github.com/jgi-kbase/IDMappingService/pull/132/files?

Comment (Xiangs18) [reply .github/workflows/manual-build.yml:1]: Previous image build GHA scripts are removed. Use docker-publish.yml now.

File: src/jgikbase/test/idmapping/service/mapper_service_test.py (hunk 2 above)
Comment (Xiangs18) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: python 3.6 -> 3.9.19

**Comment by Xiangs18** [comment]
replied in this thread: https://github.com/jgi-kbase/IDMappingService/pull/129#discussion_r1651773787

File: .github/dependabot.yml
Comment (MrCreosote) [reply .github/dependabot.yml:1]: :+1:

File: requirements.txt (hunk 1 above)
Comment (MrCreosote) [reply requirements.txt:7]: Ok, can you add an issue to find a different way to shut down the server or update the tests so it's not needed so we can update to the most recent version of werkzeug

File: src/jgikbase/test/idmapping/service/mapper_service_test.py (hunk 2 above)
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: What is the difference in the logs?

File: requirements.txt (hunk 1 above)
Comment (Xiangs18) [reply requirements.txt:7]: https://github.com/jgi-kbase/IDMappingService/issues/133

Comment (MrCreosote) [reply requirements.txt:7]: :+1:

File: src/jgikbase/test/idmapping/service/mapper_service_test.py (hunk 2 above)
Comment (Xiangs18) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: Original logstream.write.call_args_list: 
```
[call(‘{“service”: “IDMappingService”, “level”: “INFO”, “time”: 1719870386183, “source”: “jgikbase.idmapping.service.m...“, “ip”: “127.0.0.1", “method”: “GET”, “callid”: “8909603642060648", “msg”: “GET / 200 werkzeug/2.0.3"}‘),\n call(‘\n’)]
```

Current logstream.write.call_args_list:
```
[call('{"service": "IDMappingService", "level": "INFO", "time": 1718735741424, "source": "jgikbase.idmapping.service.mapper_service", "ip": "127.0.0.1", "method": "GET", "callid": "2489321465379915", "msg": "GET / 200 Werkzeug/3.0.3"}\n')]
```

Comment (Xiangs18) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: Based on above diff in logstream.write.call_args_list, need to change len and index accordingly

Comment (MrCreosote) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: I see, so it's omitting the blank line

Comment (MrCreosote) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: Ok, in that case I think you can just get rid of the lines checking that the last character of the line is a newline, that's not a valuable tests AFAICT

Comment (MrCreosote) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py:243]: Why is werkzeug listed as 3.0.3?

Comment (Xiangs18) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py]: That was log from an early commit. I can display the most recent one where werkzeug is 2.0.3.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      shell: bash
+      run: |
+        export HOMEDIR=`pwd`
+        export PYTHONPATH=$HOMEDIR/src:$PYTHONPATH
+        export IDMAP_TEST_FILE=$HOMEDIR/test.cfg
+
+        GITCOMMIT=$(git rev-parse HEAD)
+        echo "# Don't check this file into git please" > $HOMEDIR/src/jgikbase/idmapping/gitcommit.py
+        echo "commit = '${GITCOMMIT}'" >> $HOMEDIR/src/jgikbase/idmapping/gitcommit.py
+        pytest src/jgikbase/test --cov=src/jgikbase/idmapping --cov-report=xml
```
Comment (MrCreosote) [inline .github/workflows/test.yaml:70]: You should use make test here to run all the tests that travis used to run

File: Dockerfile [hunk 4]
```diff
@@ -1,6 +1,6 @@
 FROM kbase/kb_jre:latest as dockerize
 
-FROM python:3.6-alpine
+FROM python:3.9.19-alpine
```
Comment (MrCreosote) [inline Dockerfile:3]: Does the readme need to be updated based on this change?

File: src/jgikbase/test/idmapping/service/mapper_service_test.py (hunk 2 above)
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/service/mapper_service_test.py]: :+1:

File: .github/workflows/manual-build.yml
Comment (MrCreosote) [reply .github/workflows/manual-build.yml:1]: :+1:

File: Dockerfile (hunk 4 above)
Comment (Xiangs18) [reply Dockerfile:3]: yes, I will upgrade `README.md` as well as `RELEASE_NOTES.md` in a separate PR

Comment (MrCreosote) [reply Dockerfile:3]: :+1:

File: .github/workflows/test.yaml (hunk 3 above)
Comment (Xiangs18) [reply .github/workflows/test.yaml:70]: It will run the all tests that travis used to run. But you prefer `make test`?

Comment (MrCreosote) [reply .github/workflows/test.yaml:70]: It doesn't though: https://github.com/jgi-kbase/IDMappingService/blob/master/Makefile#L19-L22

Comment (Xiangs18) [reply .github/workflows/test.yaml]: `flake8 src` failed. And it passed when using travis? We can either format or ignore them.

Comment (MrCreosote) [reply .github/workflows/test.yaml]: If everything else is passing I'd say enable it and format in another PR

Comment (Xiangs18) [reply .github/workflows/test.yaml]: Only Pytest is passing. If enable any other tests (flake8, mypy, bandit), GHA fails. 
So I disabled all three and will fix them in another PR.

Comment (MrCreosote) [reply .github/workflows/test.yaml]: :+1:

**Review by MrCreosote — APPROVED** [review]
Can you make an issue for fixing flake / mypy / bandit to make sure that doesn't get lost, unless you're doing that in the next PR

**Comment by Xiangs18** [comment]
Next PR

**Pull request merged**